src/typed-geometry/functions/objects/intersection.hh

```diff
@@ -2269,4 +2327,617 @@ template <class ScalarT>
+    };
+
+    dir<3, ScalarT> ax = normalize(a.half_extents[0]);
+    dir<3, ScalarT> ay = normalize(a.half_extents[1]);
+    dir<3, ScalarT> az = normalize(a.half_extents[2]);
+
+    dir<3, ScalarT> bx = normalize(b.half_extents[0]);
+    dir<3, ScalarT> by = normalize(b.half_extents[1]);
+    dir<3, ScalarT> bz = normalize(b.half_extents[2]);
+
+    auto cross_ax_bx = cross(ax, bx);
+    auto cross_ax_by = cross(ax, by);
+    auto cross_ax_bz = cross(ax, bz);
+
+    auto cross_ay_bx = cross(ay, bx);
+    auto cross_ay_by = cross(ay, by);
+    auto cross_ay_bz = cross(ay, bz);
+
+    auto cross_az_bx = cross(az, bx);
+    auto cross_az_by = cross(az, by);
+    auto cross_az_bz = cross(az, bz);
+
+    // ax
+    if (axis_check(ax, a, b))
+        return true;
+
+    // ay
+    if (axis_check(ay, a, b))
+        return true;
+
+    // az
+    if (axis_check(az, a, b))
+        return true;
+
+    // bx
+    if (axis_check(bx, a, b))
+        return true;
+
+    // by
+    if (axis_check(by, a, b))
+        return true;
+
+    // bz
+    if (axis_check(bz, a, b))
+        return true;
+
+    // ax_bx
+    if (length(cross_ax_bx) > 0 ? axis_check(normalize(cross_ax_bx), a, b) : false)
+        return true;
+
+    // ax_by
+    if (length(cross_ax_by) > 0 ? axis_check(normalize(cross_ax_by), a, b) : false)
+        return true;
+
+    // ax_bz
+    if (length(cross_ax_bz) > 0 ? axis_check(normalize(cross_ax_bz), a, b) : false)
+        return true;
+
+    // ay_by
+    if (length(cross_ay_bx) > 0 ? axis_check(normalize(cross_ay_bx), a, b) : false)
+        return true;
+
+    // ay_by
+    if (length(cross_ay_by) > 0 ? axis_check(normalize(cross_ay_by), a, b) : false)
+        return true;
+
+    // ay_bz
+    if (length(cross_ay_bz) > 0 ? axis_check(normalize(cross_ay_bz), a, b) : false)
+        return true;
+
+    // az_bx
+    if (length(cross_az_bx) > 0 ? axis_check(normalize(cross_az_bx), a, b) : false)
+        return true;
+
+    // az_by
+    if (length(cross_az_by) > 0 ? axis_check(normalize(cross_az_by), a, b) : false)
+        return true;
+
+    // az_bz
+    if (length(cross_az_bz) > 0 ? axis_check(normalize(cross_az_bz), a, b) : false)
+        return true;
+
+    return false;
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr bool intersects(sphere<3, ScalarT> const& a, sphere<3, ScalarT> const& b)
+{
+    return distance(a.center, b.center) <= (a.radius + b.radius);
+}
+
+template <class ScalarT>
+[[nodiscard]] constexpr bool intersects(box<3, ScalarT> const& b, sphere<3, ScalarT> const& s)
+{
+    array<pos<3, ScalarT>, 8> box_vertices = vertices_of(b);
+
+    // as early-out: box vertex inside the sphere
+    for (auto const& v : box_vertices)
+    {
+        if (length(v - s.center) < s.radius)
+            return true;
+    }
+
+    array<segment<3, ScalarT>, 12> box_edges = edges_of(b);
+
+    for (auto const& e : box_edges)
+    {
+        if (intersects(e, s))
+            return true;
+    }
+
+    return false;
```

Comment:
what if the sphere is completely inside the box?

Reply:
case added